examples/iot/thing.yaml

```diff
@@ -0,0 +1,10 @@
+apiVersion: iot.aws.crossplane.io/v1alpha1
+kind: Thing
+metadata:
+  name: example-thing
+spec:
+  forProvider:
+    region: us-east-1
+    thingName: example-thing
```

Comment:
`metadata.name` has some [constraints](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names). It depends if the `thingName` accepts more relaxed constraints (like uppercase or spaces). We ran into this with `provider-gitlab`: https://github.com/crossplane-contrib/provider-gitlab/issues/41

Reply:
@janwillies  thats why i asked - doesn't know in Detail the IOT ressources

Reply:
I think it makes sense to remove `thingName` as the source of truth for external resource naming is the `external-name` annotation. And regarding the constraints of `metadata.name`, we should always be able to provision the MR with an unconstrained name by manually setting the `external-name` annotation initially, so that `managed.NameAsExternalName` initializer [does not set](https://github.com/crossplane/crossplane-runtime/blob/c72bcdd922ebdd4f24d983849c6c7fa319721af0/pkg/reconciler/managed/api.go#L55) `external-name` from `metadata.name`.

Reply:
Thanks for comments @haarchri @janwillies @ulucinar 

I ignored the thingName field in code-generator. The idea of using the annotation value to get around some name constraints makes sense.